test/print_env_info.py

```diff
@@ -0,0 +1,327 @@
+        return 'Mac OSX {} ({})'.format(version, machine())
+    if platform == 'linux':
+        # Ubuntu/Debian based
+        desc = get_lsb_version(run)
+        if desc is not None:
+            return desc
+        # Try reading /etc/*-release
+        desc = check_release_file(run)
+        if desc is not None:
+            return desc
+        return '{} ({})'.format(platform, machine())
+    # Unknown platform
+    return platform
+
+def get_gcc_version(run):
+    return run_and_parse_first_match(run, 'gcc --version', r'gcc (.*)')
+
+def get_clang_version(run):
+    return run_and_parse_first_match(run, 'clang --version', r'clang version (.*)')
+
+def get_cmake_version(run):
+    return run_and_parse_first_match(run, 'cmake --version', r'cmake (.*)')
+
+def get_nvidia_driver_version(run):
+    if get_platform() == 'darwin':
+        cmd = 'kextstat | grep -i cuda'
+        return run_and_parse_first_match(run, cmd, r'com[.]nvidia[.]CUDA [(](.*?)[)]')
+    smi = get_nvidia_smi()
+    return run_and_parse_first_match(run, smi, r'Driver Version: (.*?) ')
+
+def get_gpu_info(run):
+    if get_platform() == 'darwin':
+        if TORCH_AVAILABLE and torch.cuda.is_available():
+            return torch.cuda.get_device_name(None)
+        return None
+    smi = get_nvidia_smi()
+    uuid_regex = re.compile(r' \(UUID: .+?\)')
+    rc, out, _ = run(smi + ' -L')
+    if rc != 0:
+        return None
+    # Anonymize GPUs by removing their UUID
+    return "\n" + re.sub(uuid_regex, '', out)
+
+def get_running_cuda_version(run):
+    return run_and_parse_first_match(run, 'nvcc --version', r'V(.*)$')
+
+def get_cudnn_version(run):
+    """This will return a list of libcudnn.so; it's hard to tell which one is being used"""
+    if get_platform() == 'darwin':
+        # CUDA libraries and drivers can be found in /usr/local/cuda/. See
+        cudnn_cmd = 'ls /usr/local/cuda/lib/libcudnn*'
+    else:
+        cudnn_cmd = 'ldconfig -p | grep libcudnn | rev | cut -d" " -f1 | rev'
+    rc, out, _ = run(cudnn_cmd)
+    # find will return 1 if there are permission errors or if not found
+    if len(out) == 0 or (rc != 1 and rc != 0):
+        l = os.environ.get('CUDNN_LIBRARY')
+        if l is not None and os.path.isfile(l):
+            return os.path.realpath(l)
+        return None
+    files = set()
+    for fn in out.split('\n'):
+        fn = os.path.realpath(fn)  # eliminate symbolic links
+        if os.path.isfile(fn):
+            files.add(fn)
+    if not files:
+        return None
+    # Alphabetize the result because the order is non-deterministic otherwise
+    files = list(sorted(files))
+    if len(files) == 1:
+        return files[0]
+    result = '\n'.join(files)
+    return 'Probably one of the following:\n{}'.format(result)
+
+def get_nvidia_smi():
+    smi = 'nvidia-smi'
+    return smi
+
+def get_torchserve_version():
+    #fetch the torchserve version from version.txt file
+    with open("ts/version.txt", 'r') as file:
```

Comment:
It seems to be making an assumption that the script will be invoked from the top level directory. We should make the path relative to this script. Same for `get_torch_model_archiver`

Reply:
Fixed.